tests/simpleKVBC/TesterReplica/main.cpp

```diff
@@ -48,8 +49,11 @@ int main(int argc, char** argv) {
   }
 
   auto* dbAdapter = new concord::storage::blockchain::DBAdapter(db);
-  auto* replica = new concord::kvbc::ReplicaImp(
-      setup->GetCommunication(), setup->GetReplicaConfig(), dbAdapter, setup->GetMetricsServer().GetAggregator());
+  auto* replica = new ReplicaImp(setup->GetCommunication(),
+                                 setup->GetReplicaConfig(),
+                                 dbAdapter,
+                                 setup->GetMetricsServer().GetAggregator());
+  replica->setReplicaStateSync(new ReplicaStateSyncImp(new BlockMetadata(*replica)));
```

Comment:
Just a comment (because this doesn't seem to be very easy to address):

If someone forgets to call replica->serReplicaStateSync, the replica might throw exceptions (unless I'm missing something).

Are we setting the replicaStateSync outside the constructor because we need a pointer to replica, or because it is optional?

Reply:
Because it needs a pointer to replica.